plugintest/harness.go

```diff
@@ -0,0 +1,73 @@
+package plugintest
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+	"github.com/onsi/gomega/gexec"
+	"google.golang.org/grpc"
+)
+
+// ShutdownFunc is meant to be called to clean up resources created and in use when a plugin is started
+type ShutdownFunc func()
+
+// StartPlugin compiles and starts the backend datasource plugin at packagePath.
+// It listens on the port and passes the provided env to the plugin.
+// id should be the same as the plugin id defined in package.json
+func StartPlugin(packagePath string, id string, port int, env ...string) (*PluginClient, ShutdownFunc, error) {
+	execPath, err := gexec.Build(packagePath)
```

Comment:
this is super clever, but I am a bit skeptical of the general value since we really want to test the artifacts we build.

IIUC, this can be supported with the standalone mode, perhaps with additional (and better documented!) startup parameters

Reply:
I'm not sure I understand what you mean by `since we really want to test the artifacts we build`. Can you tell me more? 

If you're saying we want to test binaries that already exist, that's a cool idea I hadn't even considered. 

I typically use integration testing as part of the general development workflow alongside unit testing. For that use case, this is a critical piece of infrastructure because it encapsulates the steps every plugin needs to take to do its own integration tests:

1. acquire a binary
1. run the binary on a known port
1. make the grpc connection to the binary
1. provide some kind of client for the tests to use

The critical piece for in-situ development is just the building and running. I think there's a refactor to make this more flexible so it  supports connecting to an arbitrary plugin that's already running.

Reply:
> Is anything missing from the [standalone](https://github.com/grafana/grafana-plugin-sdk-go/blob/master/internal/standalone/standalone.go) startup mode? could we just add something there?

This part that was trickiest for me was setting the listen address, I think it'd be awesome to just be able to pass a flag rather than this https://github.com/grafana/grafana-plugin-sdk-go/blob/6fcd0bbc19a5c19457f2af17ce19ce0eab44d1c9/plugintest/harness.go#L54-L57

Reply:
> I think there's a refactor to make this more flexible so it supports connecting to an arbitrary plugin that's already running.

I've pushed a commit that does this.